File: src/compiler/typescript_to_rust/generators/ic_object/functions/canister_balance128.ts
import { Rust } from '../../../../../types';

export function generateIcObjectFunctionCanisterBalance128(): Rust {
    return /* rust */ `
        fn _azle_ic_canister_balance128(
            _this: &boa_engine::JsValue,
            _aargs: &[boa_engine::JsValue],
            _context: &mut boa_engine::Context
        ) -> boa_engine::JsResult<boa_engine::JsValue> {
            Ok(ic_cdk::api::canister_balance128().azle_into_js_value(_context))
        }
    `;
}

File: src/compiler/typescript_to_rust/azle_generate/src/body/ic_object/functions/canister_balance128.rs
pub fn generate() -> proc_macro2::TokenStream {
    quote::quote! {
        fn canister_balance128(
            _this: &boa_engine::JsValue,
            _aargs: &[boa_engine::JsValue],
            context: &mut boa_engine::Context
        ) -> boa_engine::JsResult<boa_engine::JsValue> {
            ic_cdk::api::canister_balance128()
                .try_into_vm_value(context)
                .map_err(|vmc_err| vmc_err.to_js_error(None))
        }
    }
}
